op-interop-mon/monitor/updater_test.go

```diff
@@ -0,0 +1,302 @@
+		initiatingInfo *supervisortypes.Identifier
+		executingInfo  eth.BlockID
+		receipts       ethtypes.Receipts
+		expectedStatus []jobStatus
+	}{
+		{
+			name: "valid log found and hash matches",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+				LogIndex:    0,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			receipts: ethtypes.Receipts{
+				{
+					Logs: []*ethtypes.Log{
+						{
+							Index: 0,
+							Data:  []byte{0x01, 0x02, 0x03},
+						},
+					},
+				},
+			},
+			expectedStatus: []jobStatus{jobStatusValid},
+		},
+		{
+			name: "log not found",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+				LogIndex:    1, // Log index 1 doesn't exist in receipts
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			receipts: ethtypes.Receipts{
+				{
+					Logs: []*ethtypes.Log{
+						{
+							Index: 0,
+							Data:  []byte{0x01, 0x02, 0x03},
+						},
+					},
+				},
+			},
+			expectedStatus: []jobStatus{jobStatusInvalid},
+		},
+		{
+			name: "log hash mismatch",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+				LogIndex:    0,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			receipts: ethtypes.Receipts{
+				{
+					Logs: []*ethtypes.Log{
+						{
+							Index: 0,
+							Data:  []byte{0x04, 0x05, 0x06}, // Different data will result in different hash
+						},
+					},
+				},
+			},
+			expectedStatus: []jobStatus{jobStatusInvalid},
+		},
+		{
+			name: "error fetching receipts",
+			initiatingInfo: &supervisortypes.Identifier{
+				ChainID:     eth.ChainIDFromUInt64(1),
+				BlockNumber: 100,
+				LogIndex:    0,
+			},
+			executingInfo: eth.BlockID{
+				Number: 200,
+				Hash:   common.HexToHash("0x123"),
+			},
+			receipts:       nil, // Will cause error in mock
+			expectedStatus: []jobStatus{jobStatusUnknown},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			updater, client := setupTestUpdater(t)
+
+			// Create a test job
+			job := &Job{
+				id:             JobID(uuid.New().String()),
+				initiating:     tt.initiatingInfo,
+				executingBlock: tt.executingInfo,
+				executingChain: eth.ChainIDFromUInt64(2),
+			}
+
+			// If we have valid receipts, calculate the expected hash
+			if len(tt.receipts) > 0 && len(tt.receipts[0].Logs) > 0 {
+				if tt.name == "log hash mismatch" {
+					// For the mismatch case, set a different hash than what we expect from the log
+					job.executingPayload = common.HexToHash("0x1234567890abcdef")
+				} else {
+					expectedHash := crypto.Keccak256Hash(supervisortypes.LogToMessagePayload(tt.receipts[0].Logs[0]))
+					job.executingPayload = expectedHash
+				}
```

Comment:
I think I'd rather hardcode correct or incorrect raw data into the test cases, than detect when the test should fail and then meddle with the test data. We can use comments to show how the raw data was generated.